src/pg_to_evalscript/javascript_processes/rename_labels.js

```diff
@@ -1,8 +1,25 @@
 function rename_labels(arguments) {
-    const {data, dimension, target, source} = arguments;
-    for (let i = 0; i < target.length; i++) {
-        const ind = data.getDimensionByName(dimension).labels.indexOf(source[i])
-        data.getDimensionByName(dimension).labels[ind] = target[i]
+  const { data, dimension, target, source = [] } = arguments;
+
+  if (!Array.isArray(data.getDimensionByName(dimension).labels) && !target) {
```

Comment:
>  If the dimension labels are not enumerated and the target parameter is not specified, the LabelsNotEnumerated exception is thrown

I would expect they meant

>  If the dimension labels are not enumerated and the *source* parameter is not specified, the LabelsNotEnumerated exception is thrown

Otherwise it doesn't make any sense? But they repeat it twice, so it must be correct. I'd be very grateful if you explain it to me, otherwise I would ask for clarification.

Reply:
Talked to Miha Kadunc about it and opened an issue: https://github.com/Open-EO/openeo-processes/issues/321

Reply:
Turns out it was indeed incorrect, *source* was meant in both cases.

--------

It would require some thinking to come up with the best way to implement this. But for now we can just say that the labels are *always* enumerated, meaning that we can always select 1st, 2nd, 3rd etc element even though the `labels` array is empty. The check should stay though.

So if `source` isn't specified, we just set `target` as the new `labels`.

Reply:
Should be fixed